File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/validation/DummyValidatorTest.java
package io.appmetrica.analytics.impl.utils.validation;

import io.appmetrica.analytics.testutils.CommonTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class DummyValidatorTest extends CommonTest {

    private final Validator<Object> mValidator = new DummyValidator<Object>();

    @Test
    public void testNull() {
        assertThat(mValidator.validate(null).isValid()).isTrue();
    }

    @Test
    public void testNonNull() {
        assertThat(mValidator.validate(new Object()).isValid()).isTrue();
    }

}

File: core-utils/src/test/java/io/appmetrica/analytics/coreutils/internal/validation/DummyValidatorTest.kt
package io.appmetrica.analytics.coreutils.internal.validation

import io.appmetrica.gradle.testutils.CommonTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class DummyValidatorTest : CommonTest() {

    private val validator = DummyValidator<Any>()

    @Test
    fun nullAndNonNull() {
        assertThat(validator.validate(null).isValid).isTrue
        assertThat(validator.validate(Any()).isValid).isTrue
    }
}
